ls1intum/Artemis
# [Bugfix] sort columns not working in submissions-dashboards

### Checklist


### Motivation and Context


The sorting buttons in the assessment-dashboards did not work. In this PR the problem is solved.

![image](https://user-images.githubusercontent.com/33342534/108605675-f4e73500-73b5-11eb-8d0c-1564519a9f8c.png)

Also a bug where no locked submissions could be cancelled is fixed here. There was a mistake in a check.
### Description

I fixed the column sorting for the modeling/file-upload/text/programming-assessment-dashboards.

### Steps for Testing

1. In a course under exercises check under "submissions" if all sort buttons work, for each exercise type.
2. Same for exam -> exercise-groups -> "Submissions"
